patterns/java/src/refactoring_guru/patterns/singleton/example/thread_safe/Singleton.java

```diff
@@ -0,0 +1,17 @@
+package refactoring_guru.patterns.singleton.example.thread_safe;
+
+public final class Singleton {
+    private static Singleton instance;
+    public static String name;
+
+    private Singleton(String name) {
+        this.name = name;
+    }
+
+    public static synchronized Singleton getInstance(String name) {
```

Comment:
There is more efficient implementation, which does not use synchronized keyword in method signature.